runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -167,37 +159,25 @@ public long append(@Nonnull Object object, @Nullable TxResolutionInfo conflictIn
                 // If we're here, we succeeded, return the acquired token
                 return tokenResponse.getSequence();
             } catch (OverwriteException oe) {
-
                 // We were overwritten, get a new token and try again.
                 log.warn("append[{}]: Overwritten after {} retries, streams {}",
-                        tokenResponse.getSequence(),
-                        x,
+                        tokenResponse.getSequence(), x,
                         Arrays.stream(streamIDs).map(Utils::toReadableId).collect(Collectors.toSet()));
 
-                TokenResponse temp;
                 if (conflictInfo == null) {
                     // Token w/o conflict info
-                    temp = runtime.getSequencerView().next(streamIDs);
+                    tokenResponse = runtime.getSequencerView().next(streamIDs);
                 } else {
-
-                    // On retry, check for conflicts only from the previous
-                    // attempt position
+                    // On retry, check for conflicts only from the previous attempt position,
+                    // otherwise the transaction will always conflict with itself.
```

Comment:
This is very subtle, can you write a unit test for it.

Reply:
@Maithem  Done, added a test.